pkg/vsphere/disk/disk_manager.go

```diff
@@ -302,8 +310,14 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 }
 
 // Detach attempts to detach a virtual disk
-func (m *Manager) Detach(op trace.Operation, d *VirtualDisk) error {
-	defer trace.End(trace.Begin(d.DevicePath))
+func (m *Manager) Detach(op trace.Operation, config *VirtualDiskConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	d, err := NewVirtualDisk(config, m.Disks)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
```

Comment:
That is what the original code does, (https://github.com/vmware/vic/blob/master/pkg/vsphere/disk/disk_manager.go#L317) Maybe @jzt can shed a light :)